modules/internal/String.chpl

```diff
@@ -1276,19 +1287,28 @@ module String {
   iter string.items() : string {
     var localThis: string = this.localize();
 
-    var i = 0;
-    while i < localThis.buffLen {
-      const curPos = localThis.buff+i;
-      const (decodeRet, cp, nBytes) = decodeHelp(buff=localThis.buff,
+    if localThis.isASCII() {
+      for i in this.byteIndices {
+        var (newBuff, allocSize) = bufferCopyLocal(localThis.buff+i,
+                                                   len=1);
+        yield chpl_createStringWithOwnedBufferNV(newBuff, 1, allocSize, 1);
+      }
+    }
+    else {
+      var i = 0;
+      while i < localThis.buffLen {
+        const curPos = localThis.buff+i;
+        const (decodeRet, cp, nBytes) = decodeHelp(buff=localThis.buff,
                                                    buffLen=localThis.buffLen,
                                                    offset=i,
                                                    allowEsc=true);
-      var (newBuf, newSize) = bufferCopyLocal(curPos, nBytes);
-      newBuf[nBytes] = 0;
+        var (newBuf, newSize) = bufferCopyLocal(curPos, nBytes);
+        newBuf[nBytes] = 0;
```

Comment:
Isn't this redundant now that bufferCopyLocal adds the terminator?

Reply:
Good catch! I did a bit more searching, couldn't find any such instance. There some cases where we add that in the String code, because we allocate a fresh buffer with `bufferAlloc` and that doesn't add null termination.

Reply:
I'll do a quick test on string tests before merging.